_pytest/config.py

```diff
@@ -1327,10 +1327,18 @@ def determine_setup(inifile, args, warnfunc=None):
     dirs = get_dirs_from_args(args)
     if inifile:
         iniconfig = py.iniconfig.IniConfig(inifile)
-        try:
-            inicfg = iniconfig["pytest"]
-        except KeyError:
-            inicfg = None
+        is_cfg_file = str(inifile).endswith('.cfg')
+        # TODO: [pytest] section in *.cfg files is depricated. Need refactoring.
+        sections = ['tool:pytest', 'pytest'] if is_cfg_file else ['pytest']
+        for section in sections:
+            try:
+                inicfg = iniconfig[section]
```

Comment:
we never deprecated `[pytest]` in custom configs, i believe we should either support both or target the features branch

Reply:
@RonnyPfannschmidt fixed

Reply:
@feuillemorte it is deprecated, but we will remove it only in 4.0. We should start issuing warnings here as well if users are using `[pytest]` in `setup.cfg` files, see:

https://github.com/pytest-dev/pytest/blob/20085542e2d55f7e6c072073a4ddb42ece5e3b07/_pytest/config.py#L1266-L1267

But we don't introduce new warnings in bug-fixes (we've broken test suites in the past because of that) so the introduction of this warning should be done in `features`, in a separate task.

Reply:
Created #3268 with that.

Reply:
@nicoddemus I see, thanks!